extensions/binding/org.eclipse.smarthome.binding.ntp.test/src/test/java/org/eclipse/smarthome/binding/ntp/test/NtpOSGiTest.java

```diff
@@ -230,15 +231,17 @@ public void testDateTimeChannelTimeZoneUpdate() {
     }
 
     @Test
-    @Ignore("https://github.com/eclipse/smarthome/issues/5224")
     public void testDateTimeChannelCalendarTimeZoneUpdate() {
         Configuration configuration = new Configuration();
         configuration.put(NtpBindingConstants.PROPERTY_TIMEZONE, TEST_TIME_ZONE_ID);
         initialize(configuration, NtpBindingConstants.CHANNEL_DATE_TIME, ACCEPTED_ITEM_TYPE_DATE_TIME, null, null);
         ZonedDateTime timeZoneIdFromItemRegistry = ((DateTimeType) getItemState(ACCEPTED_ITEM_TYPE_DATE_TIME))
                 .getZonedDateTime();
         ZoneOffset testZoneId;
-        if (timeZoneIdFromItemRegistry.getZone().getRules().isDaylightSavings(timeZoneIdFromItemRegistry.toInstant())) {
+
```

Comment:
I want to change the ZoneId instance on which getRules() method is called from timeZoneFromItemRegistry.getZone() to ZoneId.of(timeZoneInConfiguration)

Even though both of these are ZoneId instances, the second one provides more information:
printing timeZoneIdFromItemRegistry.getZone().getId() results in -07:00 and 
printing ZoneId.of(timeZoneInConfiguration).getId() results in 'America/Los_Angeles`

This information loss happens because of the formatter used in DateTimeType see this comment and the formatter can't easily be changed because it will affect PaperUI | see this comment

getRules() method returns a ZoneRules instance which has isDaylightSavings(Instant instant) method.
The way this method works is by calling the results of the getStandardOffset(Instant instant) and getOffset(Instant instant) methods for the ZoneRules instance. If both methods return different offsets, Java assumes DST is used in that instant.

As the JavaDoc states

The standard offset is the offset before any daylight saving time is applied. This is typically the offset applicable during winter.

Printing timeZoneIdFromItemRegistry.getZone().getRules().getStandardOffset(timeZoneIdFromItemRegistry.toInstant()) results in -07:00 
Printing ZoneId.of(timeZoneInConfiguration).getRules().getStandardOffset(timeZoneIdFromItemRegistry.toInstant()) results in -08:00 for timeZoneIdFromItemRegistry="2018-03-23T07:17:10.879-07:00" which is in DST in the Pacific Time Zone.

This test doesn't work because timeZoneIdFromItemRegistry.getZone() doesn't have the full information which causes timeZoneIdFromItemRegistry.getZone().getRules().isDaylightSavings(timeZoneIdFromItemRegistry.toInstant()) to give wrong result.

If there is no Timezone property available in the configuration, then the Default Timezone is used and using TimeZone.getDefault().toZoneId().getRules().isDaylightSavings(timeZoneIdFromItemRegistry.toInstant()) should return the correct result

Reply:
@lpapazow But don't we need to change the `DateTimeType` and the UI interpretation of the format, then? If the `ZoneId` information of the `DateTimeType` loss information, it can cause bugs on other places sometime, too.

Reply:
According to this [comment](https://github.com/eclipse/smarthome/pull/5226#issuecomment-375390293):

>both Paper UI and HABPanel use the AngularJS date filter to format the dates, which in turn simply uses the JavaScript Date object constructor accepting a string to deserialize them.

and the JavaScript Date constructor doesn't recognize string patterns with more specific timezone information than the offset. 

Do you think it will be easy to implement this changes in the UI?